src/commands/node/install.ts

```diff
@@ -0,0 +1,174 @@
+	}
+	if (directoryExists(installPath)) {
+		throw new Error(`Installation already exists in path ${installPath}`);
+	}
+};
+
+const parseOptions = ({ network, releaseUrl, snapshotUrl }: Flags): void => {
+	networkSupported(network);
+	isValidURL(releaseUrl);
+	isValidURL(snapshotUrl);
+};
+
+const buildOptions = async ({
+	installationPath,
+	name,
+	network,
+	releaseUrl,
+}: Flags): Promise<Options> => {
+	const installPath = LISK_INSTALL(installationPath);
+	const installDir = `${installPath}/${name}/`;
+	const latestURL = LISK_LATEST_URL(releaseUrl, network);
+	const { version, liskTarUrl, liskTarSHA256Url } = await getReleaseInfo(
+		latestURL,
+		releaseUrl,
+		network,
+	);
+
+	return {
+		installDir,
+		version,
+		liskTarUrl,
+		liskTarSHA256Url,
+	};
+};
+
+const installLisk = async (options: Flags, cacheDir: string): Promise<void> => {
+	try {
+		const {
+			installDir,
+			liskTarUrl,
+			liskTarSHA256Url,
+			version,
+		} = await buildOptions(options);
+		const { name, network, snapshotUrl, 'no-snapshot': noSnapshot } = options;
+		const LISK_RELEASE_PATH = `${cacheDir}/${LISK_TAR(version)}`;
+		const LISK_RELEASE_SHA256_PATH = `${cacheDir}/${LISK_TAR_SHA256(version)}`;
+
+		prereqCheck(installDir);
+
+		await download(liskTarUrl, LISK_RELEASE_PATH);
+		await download(liskTarSHA256Url, LISK_RELEASE_SHA256_PATH);
+		await verifyChecksum(cacheDir, LISK_TAR_SHA256(version));
+
+		if (!noSnapshot) {
+			const snapshotPath = `${cacheDir}/${LISK_DB_SNAPSHOT(name, network)}`;
+			const snapshotURL = LISK_SNAPSHOT_URL(snapshotUrl, network);
+			await download(snapshotURL, snapshotPath);
+		}
+
+		createDirectory(installDir);
+		await extract(cacheDir, LISK_TAR(version), installDir);
+	} catch (error) {
+		throw new Error(`Lisk installation failed with error: ${error}`);
+	}
+};
+
+export default class InstallCommand extends BaseCommand {
+	static description = `Install lisk software`;
+
+	static examples = [
+		'node:install',
+		'node:install --installation-path=/opt/lisk/lisk-testnet --network=testnet',
+		'node:install --snapshot-url=no',
+	];
+
+	static flags = {
+		...BaseCommand.flags,
+		network: flagParser.string({
+			char: 'n',
+			description: 'Name of the network to install(mainnet, testnet, betanet).',
+			default: 'mainnet',
+		}),
+		installationPath: flagParser.string({
+			char: 'p',
+			description: 'Path of lisk software installation.',
+			default: '~/.lisk/network',
+		}),
+		name: flagParser.string({
+			description: 'Name of the directory.',
+			default: 'main',
+		}),
+		releaseUrl: flagParser.string({
+			char: 'r',
+			description: 'URL to lisk release software.',
+			default: 'https://downloads.lisk.io/lisk',
+		}),
+		snapshotUrl: flagParser.string({
+			char: 's',
+			description: 'URL to lisk blockchain snapshot.',
+			default: 'http://snapshots.lisk.io.s3-eu-west-1.amazonaws.com/lisk',
+		}),
+		'no-snapshot': flagParser.boolean({
+			description: 'Start lisk with blockchin snapshot',
```

Comment:
`Install Lisk Core without blockchain snapshot`? also, `allowNo: false` should be added?

Reply:
i think yes, we need to add `allowNo: false`